host/class/uvc/usb_host_uvc_2/include/usb/uvc_host.h

```diff
@@ -0,0 +1,241 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include "usb/usb_host.h"
+#include "usb/usb_types_uvc.h"
+#include "esp_err.h"
+
+// Use this macros for opening a UVC stream with any VID or PID
+#define UVC_HOST_ANY_VID (0)
+#define UVC_HOST_ANY_PID (0)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct uvc_host_stream_s *uvc_host_stream_hdl_t;
+
+/**
+ * @brief Configuration structure of USB Host UVC driver
+ */
+typedef struct {
+    size_t driver_task_stack_size; /**< Stack size of the driver's task */
+    unsigned driver_task_priority; /**< Priority of the driver's task */
+    int xCoreID;                   /**< Core affinity of the driver's task */
+    bool create_background_task;   /**< When set to true, background task handling usb events is created.
+                                        Otherwise user has to periodically call uvc_host_handle_events function */
+} uvc_host_driver_config_t;
+
+/**
+ * @brief Event types of this driver
+ */
+enum uvc_host_dev_event {
+    UVC_HOST_ERROR,                  /**< USB transfer error */
```

Comment:
If there is only one option, maybe it could be reflected in the name?

Something like: 
 ```suggestion
    UVC_HOST_ERROR_TRANSFER,                  /**< USB transfer error */
```